pageserver/src/tenant/blob_io.rs

```diff
@@ -284,8 +292,12 @@ impl<const BUFFERED: bool> BlobWriter<BUFFERED> {
         srcbuf: B,
         ctx: &RequestContext,
         algorithm: ImageCompressionAlgorithm,
-    ) -> (B::Buf, Result<u64, Error>) {
+    ) -> (B::Buf, Result<u64, Error>, CompressionInfo) {
```

Comment:
nit: this interface is a bit strange -- it's not obvious what CompressionInfo means in the case that the write fails.

Probably not important if we will likely remove this metric later (this is a pretty hot code path, so once we have a sense of whether skipping incompressible data is useful, we may not want to keep these metrics around indefinitely )